presto-main/src/main/java/com/facebook/presto/operator/scalar/RegexpFunctions.java

```diff
@@ -83,10 +83,18 @@ public static Slice regexpReplace(@SqlType(StandardTypes.VARCHAR) Slice source,
     @SqlType("array<varchar>")
     public static Slice regexpExtractAll(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(RegexpType.NAME) Pattern pattern)
     {
+        return regexpExtractAll(source, pattern, 0);
+    }
+
+    @Description("group(s) extracted using the given pattern")
+    @ScalarFunction
+    @SqlType("array<varchar>")
+    public static Slice regexpExtractAll(@SqlType(StandardTypes.VARCHAR) Slice source, @SqlType(RegexpType.NAME) Pattern pattern, @SqlType(StandardTypes.BIGINT) long group)
+    {
         Matcher matcher = pattern.matcher(source.toString(UTF_8));
         ImmutableList.Builder<String> builder = ImmutableList.builder();
         while (matcher.find()) {
-            builder.add(matcher.group());
+            builder.add(matcher.group(Ints.checkedCast(group)));
```

Comment:
This throws the wrong exception for an invalid group. Extract the logic from `regexpExtract` into a helper `validateGroup` and fix the exception.